### PR TITLE
[kserve] KPIs for llm-load-test

### PR DESCRIPTION
**Pull request by drewrip**

### CODE REVIEW

**Comment by kpouget** [comment]
thanks @drewrip, looks good, can you rebase and launch a `quick` test ?

**Comment by kpouget** [comment]
@drewrip , now you can use this command to relaunch the parsing more quickly:
```
/test rhoai-plot kserve https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-psap_topsail/215/pull-ci-openshift-psap-topsail-main-rhoai-light/1762570951410585600/artifacts/light/test/artifacts/000__local_ci__run_multi_e2e_perf_test/artifacts/ci-pod-0/
```

**Comment by drewrip** [comment]
[Failed](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-psap_topsail/215/pull-ci-openshift-psap-topsail-main-rhoai-light/1762570951410585600/artifacts/light/test/artifacts/003__plots/000__projects.kserve.visualizations.kserve-llm_plots/0_matbench_parse.log) because I forgot to remove `Settings` class from the `KServeLLMPerformanceKPI` definition. I'm not sure if we'll want to include `Settings` in the payload for the KServe work but omitting for now.

**Comment by kpouget** [comment]
> [Failed](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-psap_topsail/215/pull-ci-openshift-psap-topsail-main-rhoai-light/1762570951410585600/artifacts/light/test/artifacts/003__plots/000__projects.kserve.visualizations.kserve-llm_plots/0_matbench_parse.log) because I forgot to remove Settings class from the KServeLLMPerformanceKPI definition. I'm not sure if we'll want to include Settings in the payload for the KServe work but omitting for now.

yes Drew, we want to inherit from Settings (a new class that you need to define. That's the labels we where talking about earlier in the meeting)

**Comment by kpouget** [comment]
@drewrip , can you include the KPI in the `LTS Documentation` report? 
something like this:
https://github.com/openshift-psap/topsail/blob/main/projects/notebooks/visualizations/rhods-notebooks-performance/plotting/lts_documentation.py#L73-L82
actually, I see that your KPIs are not included in the [lts payload](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-psap_topsail/215/pull-ci-openshift-psap-topsail-main-rhoai-plot/1762588341385039872/artifacts/plot/test/artifacts/lts_payload.json), you must have some glue code missing

**Comment by drewrip** [comment]
> we want to inherit from Settings 
gotcha, I didn't realize this is what was meant by labels, I thought you meant something more specific. I wasn't going to include something like `Settings` specifically in this PR since it was targeted at generating the KPIs I thought we'd discuss which labels were needed in a different PR but I suppose I could just throw the `Settings` from notebooks in there for now

**Comment by kpouget** [comment]
> I suppose I could just throw the Settings from notebooks in there for now

yes, you can start with a subset of the notebooks settings (not all of them will work here), then we'll extend the listing

File: projects/kserve/visualizations/kserve-llm/models/kpi.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+# lts.VERSION must be bumped each time a KPI is added
+#
+
+KPIs = {} # populated by the @matbench_models.KPIMetadata decorator
+
+# Throughput
+
+@matbench_models.KPIMetadata(help="Model throughput", unit="tokens/s")
+def kserve_llm_load_test_throughput(lts_payload):
+    return lts_payload.results.throughput
```
Comment (npalaska) [inline projects/kserve/visualizations/kserve-llm/models/kpi.py:13]: Hey @drewrip what does `lts_payload` here refers to? Is it the llm-load-test output json or something else?

Comment (drewrip) [reply projects/kserve/visualizations/kserve-llm/models/kpi.py:13]: This is the raw Python object that is being assembled before it is parsed by Pydantic. You can see this object being assembled in this file: https://github.com/openshift-psap/topsail/blob/main/projects/kserve/visualizations/kserve-llm/store/lts_parser.py

It collects metrics, data and info from a number of sources.

Comment (kpouget) [reply projects/kserve/visualizations/kserve-llm/models/kpi.py:13]: yes, currently it's done on multiple steps, for historical reasons:
llm-load-test output -> [matrix-benchmarking entry](https://github.com/openshift-psap/topsail/blob/693d2f4cb9375fc2f86b87ee6a6f8af28cb97e0a/projects/kserve/visualizations/kserve-llm/store/parsers.py) --> [LTS entry](https://github.com/openshift-psap/topsail/blob/693d2f4cb9375fc2f86b87ee6a6f8af28cb97e0a/projects/kserve/visualizations/kserve-llm/store/lts_parser.py) --> [KPI entries](https://github.com/openshift-psap/topsail/blob/693d2f4cb9375fc2f86b87ee6a6f8af28cb97e0a/projects/kserve/visualizations/kserve-llm/models/kpi.py)

**Comment by drewrip** [comment]
[test passed](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-psap_topsail/215/pull-ci-openshift-psap-topsail-main-rhoai-plot/1762984322136018944/artifacts/plot/test/artifacts/report_05_report:%20LTS%20Documentation.html). KPIs are being generated succesfully. @kpouget let me know what you think. Adding Labels in this [followup PR](https://github.com/openshift-psap/topsail/pull/217).

**Comment by drewrip** [comment]
[tests passing](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-psap_topsail/215/pull-ci-openshift-psap-topsail-main-rhoai-plot/1763267129517281280/artifacts/plot/test/artifacts/report_05_report:%20LTS%20Documentation.html). The raw values of the test are now exposed as a KPI as well.

**Comment by kpouget** [comment]
/approve
/lgtm
/hold
@drewrip , you should be able to merge (after the test passed) with `/hold cancel`

**Comment by drewrip** [comment]
[test passed](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-psap_topsail/215/pull-ci-openshift-psap-topsail-main-rhoai-light/1763304173379522560/artifacts/light/test/artifacts/003__plots/000__projects.kserve.visualizations.kserve-llm_plots/report_05_report:%20LTS%20Documentation.html), merging :)

**Pull request merged**